api/src/main/java/com/datastrato/gravitino/rel/expressions/partitions/Partition.java

```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright 2024 Datastrato Pvt Ltd.
+ * This software is licensed under the Apache License version 2.
+ */
+package com.datastrato.gravitino.rel.expressions.partitions;
+
+import com.datastrato.gravitino.rel.expressions.Expression;
+import java.util.Map;
+
+/** A partition represents a result of partitioning a table. */
+public interface Partition extends Expression {
```

Comment:
why Partitioin extens Expression?

Reply:
Because partitions often need to be explained by literals and field references, so I make it extends from `Expression`

Reply:
+1, 
> Because partitions often need to be explained by literals and field references, so I make it extends from Expression

I do not see any requirement for it to extend `Expression`.

Reply:
All right, it seems that we don't need it at this phase, I will remove this inheritance relationship for now.